src/core/ContactShadows.tsx

```diff
@@ -57,11 +59,23 @@ export const ContactShadows = React.forwardRef(
       const blurPlane = new THREE.Mesh(planeGeometry)
       const depthMaterial = new THREE.MeshDepthMaterial()
       depthMaterial.depthTest = depthMaterial.depthWrite = false
-      depthMaterial.onBeforeCompile = (shader) =>
-        (shader.fragmentShader = shader.fragmentShader.replace(
-          '1.0 - fragCoordZ ), opacity );',
-          '0.0 ), ( 1.0 - fragCoordZ ) * 1.0 );'
-        ))
+      depthMaterial.onBeforeCompile = (shader) => {
+        shader.uniforms = {
+          ...shader.uniforms,
+          ucolor: {
+            value: new THREE.Color(color).convertSRGBToLinear(),
+          },
+        }
+        shader.fragmentShader = shader.fragmentShader.replace(
+          `void main() {`, //
+          `uniform vec3 ucolor;\nvoid main() {`
```

Comment:
can that \n be replaced with an actual line break for readability? We are using template strings anways 🤷

Reply:
Yessir